stan/math/gpu/kernels/matrix_multiply.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef STAN_MATH_GPU_KERNELS_MATRIX_MULTIPLY_HPP
+#define STAN_MATH_GPU_KERNELS_MATRIX_MULTIPLY_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/gpu/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+// \cond
+const char* matrix_multiply_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * Matrix multiplication on the GPU
+     *
+     * @param[in] A the left matrix in matrix multiplication
+     * @param[in] B the right matrix in matrix multiplication
+     * @param[out] C the output matrix
+     * @param[in] M Number of rows for matrix A
+     * @param[in] N Number of rows for matrix B
+     * @param[in] K Number of cols for matrix A and number of rows for matrix B
+     */
+    __kernel void matrix_multiply(const __global double* A,
+                                  const __global double* B, __global double* C,
+                                  const int M, const int N, const int K) {
+      // workitem index inside the workgroup
+      const int workgroup_row = get_local_id(0);
+      const int workgroup_col = get_local_id(1);
```

Comment:
Should these all be changed to size_t? and would a better name here be workitem_row/col?

Reply:
Maybe something like local_row, local_col would be better. These are indices inside a workgroup/thread block, thats why I put that prefix there.

Reply:
Agree on the size_t comment. I will change this together with the changes to thread/thread block naming conventions as I will have to go through all the kernels anyway.